etc/iso.c

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#include "iso.h"
+#include <string.h>
+#include "kerneltypes.h"
+
+#define PVD_OFFSET 156
+#define ISO_BLOCK 2048
+
+struct directory_record {
+	uint8_t length_of_record;
+	uint8_t length_of_ext_record;
+	char loc_of_ext[8];
+	char data_length[8];
+	char rec_date_time[7];
+	char file_flags[1];
+	char file_flags_interleaved[1];
+	uint8_t interleave_gap_size;
+	char vol_seq_num[4];
+	uint8_t len_identifier;
+	char file_identifier[2048];
+};
+
+void print_n_tabs(int n);
+void print_dr_name(struct directory_record *DR);
+int hex_to_int(char *src, int len);
+struct directory_record get_directory_record(FILE *fp);
+int is_dir(int flags);
+int is_valid_record(struct directory_record *DR);
+void print_subdirectory(FILE *fp, char *loc_of_subdir, char *loc_of_parent, int depth);
+int have_equal_extents(char *first, char *second);
+
+void print_file_system(FILE *fp) {
+	fseek(fp, (16 * ISO_BLOCK) + PVD_OFFSET, SEEK_SET);
+	struct directory_record DR;
+	char loc_of_parent[8];
+	DR = get_directory_record(fp);	// This is the DR for the PVD
+	if (is_valid_record(&DR)) {
+		print_dr_name(&DR);
+		printf("\n");
+	}
+	memcpy(loc_of_parent, DR.loc_of_ext, 8);
+	print_subdirectory(fp, DR.loc_of_ext, loc_of_parent, 1);
+}
+
+void print_subdirectory(FILE *fp, char *loc_of_subdir, char *loc_of_parent, int depth) {
+	fseek(fp, ISO_BLOCK * hex_to_int(loc_of_subdir + 4, 4), SEEK_SET);	// Go to extent of subdir
+	struct directory_record DR;
```

Comment:
variable names should be lower case. Upper Case is reserved for defined constants